packages/langium/src/utils/uri-utils.ts

```diff
@@ -16,6 +16,8 @@ export namespace UriUtils {
     export const joinPath = Utils.joinPath;
     export const resolvePath = Utils.resolvePath;
 
+    const isWindows = process?.platform === 'win32';
```

Comment:
This is not portable. Here's a good reference for how to make it work both in Node.js and browser:

https://github.com/eclipse-theia/theia/blob/master/packages/core/src/common/os.ts

Reply:
Additionally, I'd be in favor of calling `URI.parse` on the input in case the input is a string. It isn't obvious we are expecting a path, and should much rather expect a stringified URI, like we're doing in the other methods.